File: regular_assignments/day_013/element_removal/solution.rb
class Solution
  def call
    solve([2,1]) # 4
    solve([5]) # 5
    solve([8,0,10]) # 26
  end

  def solve(arr)
    arr = arr.sort.reverse
    cost = 0
    i = 0
    while i < arr.length
      cost += sumation(arr[i..-1])
      i += 1
    end
    cost
  end

  def sumation(arr)
    sum = 0
    i = 0
    while i < arr.length
      sum+= arr[i]
      i += 1
    end
    sum
  end
end

# puts '-' * 100
# puts Solution.new.solve([2,1])
# puts '-' * 100
# puts Solution.new.solve([5])
# puts '-' * 100
# puts Solution.new.solve([8,0,10])

File: regular_assignments/day_013/element_removal/solution.go
package elementRemovalCost

import (
	"slices"
	"sort"
)

func main() {
	solve([]int{2, 1})
	solve([]int{5})
	solve([]int{8, 0, 10})
}

func solve(arr []int) int {
	sort.Ints(arr)
	slices.Reverse(arr)

	cost := 0
	for i := 0; i < len(arr); i++ {
		for j := i; j < len(arr); j++ {
			cost += arr[j]
		}
	}
	return cost
}
